### PR TITLE
[UI] - Layout, breadcrumbs, pageloader and styling refactoring

### PR DESCRIPTION
**Pull request by kuby**
- `Breadcrumbs` component
- `PageLoader` component
- Added `Helmet` for dynamically update the app title
- AuthConfig is now fetched from the `auth_config.json` file which is inside the `public` folder
- Refactored styling: created color and size constants, refactored the collapsed app state and classes
- `Layout` component for generating a full width or split layout [integrated Breadcrumbs, PageLoader, Helmet]
- Integrated PropTypes
- Added lodash
- Created `things` slice (reducer, action, selector)

### CODE REVIEW

**Comment by kuby** [comment]
@jkralik FYI, this MR contains the dynamic `auth_config.json` fetching.

File: http-gateway/web/src/common/styles/default.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
+@import './colors.scss';
+@import './sizes.scss';
+
+html,
+body,
+input,
+textarea,
+select {
+  font-family: 'Muli', sans-serif;
```
Comment (oskarbauer) [inline http-gateway/web/src/common/styles/default.scss:9]: Muli is a licensed adobe font and a "free" font from font squirrel?

File: http-gateway/web/src/components/layout/layout.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+import { layoutTypes } from './constants'
+
+const { FULL_PAGE, SPLIT } = layoutTypes
+
+/**
+ * Basic layout component.
+ * @param {Element} header - Elements to be rendered inline with the breadcrumbs, but justified to the end.
+ * @param {Array} breadcrumbs - Breadcrumbs to be rendered.
+ * @param {Boolean} loading - Dispaly's a loader below status bar.
```
Comment (oskarbauer) [inline http-gateway/web/src/components/layout/layout.js]: typo

File: http-gateway/web/src/common/styles/default.scss (hunk 1 above)
Comment (kuby) [reply http-gateway/web/src/common/styles/default.scss:9]: It appears so, I will investigate further.

File: http-gateway/web/src/components/layout/layout.js (hunk 2 above)
Comment (kuby) [reply http-gateway/web/src/components/layout/layout.js]: Right, thanks.

**Review by oskarbauer — APPROVED** [review]

**Pull request merged**